### PR TITLE
Upload escher to artifactory.

### PR DESCRIPTION
**Pull request by Talina06**
Path uploaded to:
```
$ aws s3 ls s3://<s3-bucket>/escher/
                           PRE linux-v2.0.2/
                           PRE osx-v2.0.2/
```
```
$ aws s3 ls s3://<s3-bucket>/escher/linux-v2.0.2/
                           PRE plugins/
                           escher
```
```
$ aws s3 ls s3://<s3-bucket>/escher/linux-v2.0.2/plugins/
 artifactory.so
 kv.so
 run.so
```
Ex: if v2.0.2 is the latest version. Same will be pushed to github releases and uploaded to artifactory.

### CODE REVIEW

**Comment by Talina06** [comment]
@jdk2588 Please review now. Incorporated changes. The developer will have to do 2 things: Bump up the version in the code. and tag the commit with the same version while pushing.

File: .travis.yml [hunk 1]
```diff
@@ -31,3 +31,15 @@ deploy:
...
     tags: true
+
+  provider: s3
+  skip_cleanup: true
+  access_key_id: ${AWS_ACCESS_KEY}
+  secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+  region: ap-south-1
+  bucket: "sre-inventory"
+  local_dir: output/${TRAVIS_BUILD_NUMBER}/${TRAVIS_OS_NAME}/amd64/
+  upload-dir: escher/${TRAVIS_OS_NAME}-${TRAVIS_TAG}
```
Comment (jdk2588) [inline .travis.yml]: TRAVIS_TAG is passed from where?

Comment (jdk2588) [reply .travis.yml]: Also, the version can't be read from the utility? Like if I do 
```escher version```  and the output is read? 
Also, while merging to the master. Can't Travis add a tag and release to Github? 
You will have to check.

Comment (Talina06) [reply .travis.yml]: Working on it. will let you know if this works.

**Comment by Talina06** [comment]
@jdk2588 I've done the following:

2 travis builds run: 1 for each OS type: osx and linux.
Each build does this:
builds escher. 
captures the `escher --version`
generates a git tag for `$os-$tag`
and deploys on github releases and s3 when merged to master for the above git tag.

**Comment by Talina06** [comment]
@mohandutt134 @jdk2588 @meson10 Please review this PR.

File: cli.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ var (
 	pluginDir  = defaultPluginDir()
 )
 
+const Version = "v1.0.0"
```
Comment (meson10) [inline cli.go]: Why is there a "v" in version number?
Also, It's not really version 1.0.0

File: .travis.yml [hunk 3]
```diff
@@ -20,14 +20,29 @@ script:
   - cp dist/${TRAVIS_OS_NAME}/escher output/${TRAVIS_BUILD_NUMBER}/${TRAVIS_OS_NAME}/amd64/
   - cp dist/${TRAVIS_OS_NAME}/*.so output/${TRAVIS_BUILD_NUMBER}/${TRAVIS_OS_NAME}/amd64/plugins/
 
+before_deploy:
+  - export TRAVIS_TAG=${TRAVIS_OS_NAME}-$(dist/${TRAVIS_OS_NAME}/escher --version 2>&1 | tail -1)
+  - if [ -z $(git fetch --tags && git tag | grep ${TRAVIS_TAG}) ]; then git tag ${TRAVIS_TAG} -a -m "Generated build for ${TRAVIS_TAG}" && git push -q https://${GITHUB_OAUTH_TOKEN}@github.com/tsocial/escher --tags; fi
```
Comment (meson10) [inline .travis.yml]: This is the same code that I had pointed out as incorrect. Last time.
Release != Version

File: cli.go (hunk 2 above)
Comment (Talina06) [reply cli.go]: editing

File: .travis.yml (hunk 3 above)
Comment (Talina06) [reply .travis.yml]: Fixed, please review again. :pray:

**Pull request merged**